### PR TITLE
Create Brand page and All Brands page. (#180)

### PR DESCRIPTION
**Pull request by damdiana**

### CODE REVIEW

File: db_structure.sql [hunk 1]
```diff
@@ -1,6 +1,6 @@
 CREATE TABLE Products (
     id VARCHAR(10) PRIMARY KEY,
-    brand_id VARCHAR(255),
+    brand_id INT,
```
Comment (iampava) [inline db_structure.sql:3]: How are you planning to migrate the data in production? Do you plan to just delete everything and then add back the items?

File: src/app/api/brands/[brandId]/route.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
+import { response200, response404, response500 } from "@/app/utils";
+import { getBrand } from "@/model/BrandModel";
+import { getProducts } from "@/model/ProductModel";
+
+export async function GET(
```
Comment (iampava) [inline src/app/api/brands/[brandId]/route.ts:5]: If we're not using this route, let's not add it now.

I don't see any `BrandAPI` so...

File: src/app/api/brands/route.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import { response200, response500 } from "@/app/utils";
+import { getBrands } from "@/model/BrandModel";
+
+export async function GET(request: Request) {
```
Comment (iampava) [inline src/app/api/brands/route.ts:4]: Same here. Is this route used? If not, let's delete

File: src/model/BrandModel.ts [hunk 4]
```diff
@@ -0,0 +1,37 @@
+import { getPostgresClient } from "@/services/server/database";
+import { Brand } from "@/services/types";
+
+async function getBrand(brandId: string): Promise<Brand | undefined> {
```
Comment (iampava) [inline src/model/BrandModel.ts:4]: Shouldn't `brandId` be `number`? You modified it to `INT` in this commit, right? Talking about the SQL column.

File: src/app/brands/[brandId]/page.tsx [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+
+  try {
+    [products, brand] = await Promise.all([
+      getProducts({
+        brandId: params.brandId,
+      }),
+      getBrand(params.brandId),
+    ]);
+  } catch (err) {
+    console.error(err);
```
Comment (iampava) [inline src/app/brands/[brandId]/page.tsx:45]: If this fails we just `console.error`? Shouldn't we show some information to the user?

File: src/app/brands/[brandId]/page.tsx [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+      </BrandPage>
+    );
+  }
+
+  if (user !== undefined) {
+    try {
+      let favorites = await getFavorites(user.id);
+      favoriteProductsIds = favorites.map((favorite) => favorite.id);
+    } catch (err) {
+      console.error("Failed to show favorite products", err);
```
Comment (iampava) [inline src/app/brands/[brandId]/page.tsx:67]: Technically we should show an error message here as well. Try and think of a way in which we don't duplicate a lot of code, but if we fail here we show a `<p>` below the one with the brand name

```
 <p className="font-bold text-center text-xl"> {brand.name}</p>
```

File: src/app/brands/page.tsx [hunk 7]
```diff
@@ -0,0 +1,26 @@
+import { HEADER_NAV } from "@/Constants";
+import Header from "@/components/Header/Header";
+import { getBrands } from "@/model/BrandModel";
+import getUserServerSide from "@/services/server/UserService";
+import Link from "next/link";
+
+export default async function Page() {
+  const brands = await getBrands();
+  const user = await getUserServerSide();
```
Comment (iampava) [inline src/app/brands/page.tsx:9]: Why not `Promise.all`? Also, something fails, what do we show?

File: src/app/brands/page.tsx [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+
+  return (
+    <div>
+      <Header nav={HEADER_NAV} user={user} />
+      <h2 className="font-bold text-center text-xl m-2"> All brands </h2>
+      <div className="grid card-grid-cols-3 mt-4 gap-4">
+        {brands.map((brand) => {
+          return (
```
Comment (iampava) [inline src/app/brands/page.tsx:17]: Nitpick: let's remove the return and have the arrow function return directly

**Review by iampava — CHANGES_REQUESTED** [review]

File: db_structure.sql (hunk 1 above)
Comment (damdiana) [reply db_structure.sql:3]: I will create a script to update the data type of the brand_id: 
```
ALTER TABLE brands 
ALTER COLUMN brand_id TYPE INT;
```

File: src/app/brands/[brandId]/page.tsx [hunk 9]
```diff
@@ -0,0 +1,94 @@
...
+    title: brand.name,
+  };
+}
+
+export default async function Page({
+  params,
+}: {
+  params: { brandId: number };
+}) {
+  const user = await getUserServerSide();
```
Comment (iampava) [inline src/app/brands/[brandId]/page.tsx:33]: Why not add this in `Promise.all` as well?

**Review by iampava — APPROVED** [review]
One small thing then merge!

**Pull request merged**